lms/djangoapps/course_api/blocks/transformers/tests/test_extra_fields.py

```diff
@@ -0,0 +1,49 @@
+"""
+Tests for ExtraFieldsTransformer.
+"""
+from django.conf import settings
+
+# pylint: disable=protected-access
+from openedx.core.djangoapps.content.block_structure.factory import BlockStructureFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import SampleCourseFactory
+
+from ..extra_fields import ExtraFieldsTransformer
+
+
+class TestExtraFieldsTransformer(ModuleStoreTestCase):
+    """
+    Test proper behavior for ExtraFieldsTransformer
+    """
+    shard = 4
+
+    def setUp(self):
+        super(TestExtraFieldsTransformer, self).setUp()
+        self.course_key = SampleCourseFactory.create().id
+        self.course_usage_key = self.store.make_course_usage_key(self.course_key)
+        self.block_structure = BlockStructureFactory.create_from_modulestore(self.course_usage_key, self.store)
+
+        self.old_extra_fields = settings.FEATURES.get('COURSE_BLOCKS_API_EXTRA_FIELDS')
```

Comment:
Things aren't super consistent just now but I think `settings.FEATURES` should mostly be used for feature flags (boolean on/off values only). I would recommend moving this out of `FEATURES` so it's just `settings. COURSE_BLOCKS_API_EXTRA_FIELDS`).

Then you can use the standard Django way to override settings - https://docs.djangoproject.com/en/3.0/topics/testing/tools/#overriding-settings

Reply:
Oh, I wasn't aware of the standard way to override settings! That'll be extremely helpful, thanks a lot :slightly_smiling_face:!